rust/feature-flags/src/flag_matching.rs

```diff
@@ -805,6 +824,85 @@ impl FeatureFlagMatcher {
         }
     }
 
+    pub async fn evaluate_cohort_filters(
+        &self,
+        filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        Box::pin(self.evaluate_potentially_nested_cohort_filters(filters, target_properties)).await
```

Comment:
`evaluate_cohort_filters` is called recursively, hence me pinning the memory here.  See [this issue](https://github.com/rust-lang/rust/pull/117703/).  Gotta love async programming in Rust.